lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,11 +108,149 @@ class RsqrtOpConversion : public OpConversionPattern<tosa::RsqrtOp> {
   }
 };
 
+template <typename SrcOp>
+SmallVector<Value, 2>
+create_broadcast_op_if_needed(SrcOp &srcOp, ArrayRef<Value> operands,
+                              ConversionPatternRewriter &rewriter) {
+  // TOSA allows implicit broadcasting, so we need to insert broadcast_in_dim
+  // ops if necessary, e.g.:
+  // tensor<8xf32>     -> tensor<1x4x4x8xf32> Broadcast dims = (3)
+  // tensor<4x8xf32>   -> tensor<1x4x4x8xf32> Broadcast dims = (2, 3)
+  // tensor<4x4x8xf32> -> tensor<1x4x4x8xf32> Broadcast dims = (1, 2, 3)
+
+  StringRef broadcastFuncName = "emitc::broadcast_in_dim";
+  StringAttr broadcastCallee = rewriter.getStringAttr(broadcastFuncName);
+
+  Value output = srcOp.getResult();
+  auto opOutputShape = output.getType().cast<RankedTensorType>().getShape();
+  auto opOutputRank = output.getType().cast<RankedTensorType>().getRank();
+  SmallVector<Value, 2> broadcastedOperands;
+  broadcastedOperands.push_back(operands[0]);
+  broadcastedOperands.push_back(operands[1]);
+  for (size_t i = 0; i < operands.size(); i++) {
```

Comment:
You may want to change this to
```suggestion
  for (size_t i = 0; i < operands.size(); ++i) {
```

Reply:
I'm not quite sure why GitHub still shows the old code here. I already switched to post increment in this loop.

Reply:
Have you pushed your changes?